### PR TITLE
Move option parsing to its own file and decompose functions in inject.js

### PR DESCRIPTION
**Pull request by rachel-fenichel**

### CODE REVIEW

File: core/inject.js [hunk 1]
```diff
@@ -26,6 +26,7 @@
 
 goog.provide('Blockly.inject');
 
+goog.require('Blockly.Options');
```
Comment (NeilFraser) [inline core/inject.js]: Requires should be alphabetical.  Swap with 'Css'.

File: core/options.js [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+goog.provide('Blockly.Options');
+
+
+/**
+ * Parse the user-specified options, using reasonable defaults where behaviour
+ * is unspecified.
+ * @param {!Object} options Dictionary of options.  Specification:
+ *   https://developers.google.com/blockly/installation/overview#configuration
+ * @return {!Blockly.Options} Object containing a dicitonary of normalized
```
Comment (NeilFraser) [inline core/options.js]: This function does not return anything.  It is a @constructor.

File: core/options.js [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+  this.hasSounds = hasSounds;
+  this.hasCss = hasCss;
+  this.languageTree = languageTree;
+  this.gridOptions = Blockly.Options.parseGridOptions_(options);
+  this.zoomOptions = Blockly.Options.parseZoomOptions_(options);
+  this.enableRealtime = enableRealtime;
+  this.realtimeOptions = realtimeOptions;
+};
+
+Blockly.Options.prototype.parentWorkspace = null;
```
Comment (NeilFraser) [inline core/options.js]: Missing JSDoc.

File: core/options.js [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+ * if unset.
+ * @return {Object} Contains size an position metrics, or null.
+ */
+Blockly.Options.prototype.getMetrics = function() { return null; };
+
+/**
+ * Parse the user-specified zoom options, using reasonable defaults where
+ * behaviour is unspecified.  See zoom documentation:
+ *   https://developers.google.com/blockly/installation/zoom
+ * @param {!Object} options Dictionary of options options.
```
Comment (rachel-fenichel) [inline core/options.js]: Just noticed repeated "options"; will remove.

**Comment by NeilFraser** [comment]
LGTM

**Pull request merged**